debian/changelog

```diff
@@ -2,8 +2,9 @@ cloud-init (23.4.1-0ubuntu1~20.04.3) UNRELEASED; urgency=medium
 
   * d/cloud-init.maintscript: remove /etc/cloud/clean.d/README
   * d/cloud-init.logrotate: add logrotate config for cloud-init
+  * Upstream snapshot based on upstream/main at b08193b3.
```

Comment:
Your new_upstream_snapshot also contains quilt patch refreshes  to `debian/patches/*`  that aren't documented in debian/changelog. Other than that this looks great. +1